### PR TITLE
E2E test to verify workflow for deleting PVC when PVC is already in use by a POD

### PR DESCRIPTION
**Pull request by divyenpatel**
**What this PR does / why we need it**:
This PR updates e2e test to add more verification when PVC is deleted before the Pod.
Added verification for volume attachment, and volume accessibility in the pod.

**Which issue this PR fixes** 
fixes #https://github.com/vmware/kubernetes/issues/279

**Special notes for your reviewer**:
Please review: @BaluDontu @rohitjogvmw 

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: test/e2e/storage/persistent_volumes-vsphere.go [hunk 1]
```diff
@@ -150,6 +150,20 @@ var _ = SIGDescribe("PersistentVolumes:vsphere", func() {
...
+		Expect(isVolumeAttached).To(BeTrue())
+
+		pvs := make([]*v1.PersistentVolume, 1)
+		pvs[0] = pv
+		By("Verify the volume is accessible and available in the pod")
+		verifyVSphereVolumesAccessible(pod, pvs, vsp)
+		framework.Logf("Verified that Volume is accessible in the POD after deleting PV claim")
+
 		By("Deleting the Pod")
 		framework.ExpectNoError(framework.DeletePodWithWait(f, c, clientPod), "Failed to delete pod ", clientPod.Name)
```
Comment (BaluDontu) [inline test/e2e/storage/persistent_volumes-vsphere.go]: After you delete the pod, verify that PV is deleted.

File: test/e2e/storage/persistent_volumes-vsphere.go [hunk 2]
```diff
@@ -150,6 +150,20 @@ var _ = SIGDescribe("PersistentVolumes:vsphere", func() {
...
+		isVolumeAttached, verifyDiskAttachedError := verifyVSphereDiskAttached(vsp, pv.Spec.VsphereVolume.VolumePath, types.NodeName(pod.Spec.NodeName))
+		Expect(verifyDiskAttachedError).NotTo(HaveOccurred())
+		Expect(isVolumeAttached).To(BeTrue())
+
+		pvs := make([]*v1.PersistentVolume, 1)
+		pvs[0] = pv
+		By("Verify the volume is accessible and available in the pod")
+		verifyVSphereVolumesAccessible(pod, pvs, vsp)
+		framework.Logf("Verified that Volume is accessible in the POD after deleting PV claim")
+
```
Comment (BaluDontu) [inline test/e2e/storage/persistent_volumes-vsphere.go]: Can you check that PV has failed to delete as pod is still running.

**Comment by divyenpatel** [comment]
@BaluDontu Addressed review comments.

**Review by BaluDontu — APPROVED** [review]

**Comment by BaluDontu** [comment]
LGTM.

**Review by rjog — COMMENTED** [review]
/lgtm

**Comment by divyenpatel** [comment]
Reviewed Internally. Generating external PR.

**Pull request merged**